controller_manager/include/controller_manager/controller_manager.hpp

```diff
@@ -349,6 +354,10 @@ class ControllerManager : public rclcpp::Node
   };
 
   SwitchParams switch_params_;
+  //Per controller update rate support
+  int update_loop_counter_ = 0;
+  int update_rate_ = 100;
```

Comment:
This should probably be `unsigned`-type because negative values are not sensible.